### ISSUE TITLE
Feature Request For routes instead of port number

### ISSUE THREAD

**Issue by Aditya-R-K** [getumbrel/umbrel#1900]
Instead of giving port number, can we add feature to provide routes for example for nextcloud we can provide http://umbrel.local/adguard so it will internally redirect to http:umbrel.local:8085

**Comment by JoseMoranUrena523** [comment]
Having it by a port number gives a little bit tougher security (because of instead of going /adguard and cracking the password, you have to find the port number which could be a bit tricky). The public isn't supposed to be able to access the apps.

**Comment by Magnusretrotech** [comment]
This is such a good idea. I want this really badly.

**Comment by Aditya-R-K** [comment]
> Having it by a port number gives a little bit tougher security (because of instead of going /adguard and cracking the password, you have to find the port number which could be a bit tricky). The public isn't supposed to be able to access the apps.

individual users can configure any name for user can configure mydns, adguarddns, adguard, adblocker, customdns

**Comment by Emilien-Lambert** [comment]
This is a bad idea, using routes like [http://umbrel.local/adguard](http://umbrel.local/adguard) instead of port numbers can cause conflicts, as many applications have their own routing paths. This approach is not standard and can lead to routing issues.

The preferred method is to use subdomains, such as [http://adguard.umbrel.local/](http://adguard.umbrel.local/). Subdomains prevent routing conflicts, maintain clean architecture, and enhance security through the Same Origin Policy. They also offer SEO benefits by being treated as separate entities by search engines.
In short, while using routes might seem simpler, subdomains are a more robust and scalable solution. 

As a bonus, you can achieve this using the Cloudflare Tunnel app, which is highly secure and easy to use. This allows you to set up a subdomain like [https://subdomain.mydomain.com](https://subdomain.mydomain.com), with the option to make the route public or private just for you. See [cloudflared](https://apps.umbrel.com/app/cloudflared)

**Comment by daallgeier** [comment]
Not a fan of the initially suggested option using route either. But a built in service.umbrel.local nginx setup would be dope.

**Comment by kerbless** [comment]
> The preferred method is to use subdomains,
+1
